platform/include/platform/Span.h

```diff
@@ -665,6 +704,44 @@ struct Span<ElementType, SPAN_DYNAMIC_EXTENT> {
     Span(element_type (&elements)[Count]):
         _data(elements), _size(Count) { }
 
+    /**
+     * Construct a Span from the reference to a std::array.
+     *
+     * @param array Reference to the array viewed.
+     *
+     * @post a call to size() returns Count, and data() returns a
+     * pointer to elements as if by array.data().
+     */
+    template<typename OtherElementType, size_t Count>
+    Span(std::array<OtherElementType, Count> &array):
+        _data(array.data()), _size(Count)
+    {
+        MBED_STATIC_ASSERT(
```

Comment:
This has been open for ages, not sure what's holding it up: https://github.com/ARMmbed/mbed-os/pull/13085

I'd suggest using `static_assert` directly in this new code.